hs/app/reach-cli/Main.hs

```diff
@@ -0,0 +1,352 @@
+  cmd "echo" (i <> ":") (Output s)
+
+
+--------------------------------------------------------------------------------
+cmdWhoami :: Mod CommandFields Cli
+cmdWhoami = command "whoami" $ info (pure Whoami) fullDesc
+
+
+whoami :: Script ()
+whoami = cmd "docker" "info" "--format" "{{.ID}}"
+  |> (stdError, T.unpack "/dev/null")
+
+
+--------------------------------------------------------------------------------
+cmdNumericVersion :: Mod CommandFields Cli
+cmdNumericVersion = command "numeric-version" $ info (pure NumericVersion) fullDesc
+
+
+numericVersion :: Script ()
+numericVersion = undefined
+
+
+--------------------------------------------------------------------------------
+cmdReactDown :: Mod CommandFields Cli
+cmdReactDown = command "react-down" $ info (pure ReactDown) fullDesc
+
+
+reactDown :: Script ()
+reactDown = undefined
+
+
+--------------------------------------------------------------------------------
+cmdRpcServerDown :: Mod CommandFields Cli
+cmdRpcServerDown = command "rpc-server-down" $ info (pure RpcServerDown) fullDesc
+
+
+rpcServerDown :: Script ()
+rpcServerDown = undefined
+
+
+--------------------------------------------------------------------------------
+cmdUnscaffold :: Mod CommandFields Cli
+cmdUnscaffold = command "unscaffold" $ info (pure Unscaffold) fullDesc
+
+
+unscaffold :: Script ()
+unscaffold = undefined
+
+
+--------------------------------------------------------------------------------
+-- TODO better header
+header' :: String
+header' = "https://reach.sh"
+
+
+cmds :: ParserInfo Cli
+cmds = info (hsubparser cs <|> hsubparser hs <**> helper) im where
+  im = header header' <> fullDesc
+  cs = cmdCompile
+    <> cmdClean
+    <> cmdInit
+    <> cmdRun
+    <> cmdDown
+    <> cmdScaffold
+    <> cmdReact
+    <> cmdRpcServer
+    <> cmdRpcRun
+    <> cmdDevnet
+    <> cmdUpgrade
+    <> cmdUpdate
+    <> cmdDockerReset
+    <> cmdVersion
+    <> cmdHashes
+    <> cmdHelp
+  hs = internal
+    <> commandGroup "hidden subcommands"
+    <> cmdNumericVersion
+    <> cmdReactDown
+    <> cmdRpcServerDown
+    <> cmdUnscaffold
+    <> cmdWhoami
+
+
+main :: IO ()
+main = customExecParser (prefs showHelpOnError) cmds >>= \case
+  Clean m i      -> sh $ clean m i
+  Compile        -> sh compile
+  Devnet         -> sh devnet
+  DockerReset    -> sh dockerReset
+  Down           -> sh down
+  Hashes         -> sh hashes
+  Help           -> sh help'
+  Init           -> sh init'
+  React          -> sh react
+  RpcRun         -> sh rpcRun
+  RpcServer      -> sh rpcServer
+  Run            -> sh run'
+  Scaffold       -> sh scaffold
+  Update         -> sh update
+  Upgrade        -> sh upgrade
+  Version        -> sh version
+
+  -- Hidden
+  NumericVersion -> sh numericVersion
+  ReactDown      -> sh reactDown
+  RpcServerDown  -> sh rpcServerDown
+  Unscaffold     -> sh unscaffold
+  Whoami         -> sh whoami
+
+ where
+  sh f = T.putStrLn . script $ do
```

Comment:
I think the commands should be in `IO` and return scripts

Reply:
I was actually already planning to do something about all the repetition but I usually focus on consolidation once more of the body has taken shape.

How do you feel about this way instead?

Reply:
Yea, I like it better. Although I think I'd probably put the name with the code